```
Error running mint automation
+ mkdir -p 11647-ed4281f/gopath/src/github.com/minio
+ git -C 11647-ed4281f/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 11647-ed4281f/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 11647-ed4281f/gopath/src/github.com/minio/minio fetch --quiet minio pull/11647/head:pr11647
+ git -C 11647-ed4281f/gopath/src/github.com/minio/minio checkout --quiet pr11647
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/11647-ed4281f/gopath
+ make -C 11647-ed4281f/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/minio/minio-go/v7 v7.0.10
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/shirou/gopsutil v3.20.11+incompatible
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/dswarbrick/smart v0.0.0-20190505152634-909a45200d6d
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/willf/bitset v1.1.11
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/philhofer/fwd v1.1.1
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/minio/highwayhash v1.0.1
go: downloading github.com/klauspost/compress v1.11.7
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/prometheus/procfs v0.2.0
go: downloading golang.org/x/net v0.0.0-20201216054612-986b41b23924
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/streadway/amqp v1.0.0
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/klauspost/cpuid/v2 v2.0.4
go: downloading github.com/prometheus/client_golang v1.8.0
go: downloading github.com/tinylib/msgp v1.1.3
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/lib/pq v1.8.0
go: downloading github.com/klauspost/reedsolomon v1.9.11
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/minio/sio v0.2.1
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/miekg/dns v1.1.35
go: downloading golang.org/x/text v0.3.3
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading golang.org/x/crypto v0.0.0-20201124201722-c8d3bf9c5392
go: downloading golang.org/x/sys v0.0.0-20210119212857-b64e53b001e4
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: downloading github.com/nats-io/nkeys v0.2.0
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/olivere/elastic/v7 v7.0.22
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/pierrec/lz4 v2.5.2+incompatible
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading google.golang.org/protobuf v1.23.0
go: downloading github.com/gorilla/handlers v1.5.1
go: downloading go.etcd.io/etcd v0.0.0-20201125193152-8a03d2e9614b
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/go-ldap/ldap v3.0.2+incompatible
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/eapache/queue v1.1.0
go: downloading google.golang.org/grpc v1.26.0
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/fatih/color v1.10.0
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/felixge/httpsnoop v1.0.1
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/nats-io/stan.go v0.7.0
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.0
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/nats-io/jwt v1.1.0
go: downloading github.com/minio/md5-simd v1.1.1
go: downloading github.com/gomodule/redigo v1.8.3
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/tidwall/gjson v1.6.7
go: downloading github.com/golang/groupcache v0.0.0-20190702054246-869f871628b6
go: downloading github.com/gogo/protobuf v1.3.1
go: downloading go.uber.org/zap v1.13.0
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/tidwall/pretty v1.0.2
go: downloading github.com/tidwall/match v1.0.3
go: downloading github.com/hashicorp/vault/api v1.0.4
go: downloading go.uber.org/atomic v1.5.0
go: downloading go.uber.org/multierr v1.3.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading golang.org/x/time v0.0.0-20191024005414-555d28b269f0
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/coreos/pkg v0.0.0-20160727233714-3ac0863d7acf
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/coreos/go-semver v0.2.0
go: downloading github.com/coreos/go-systemd v0.0.0-20180511133405-39ca1b05acc7
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/ryanuber/go-glob v1.0.0
# github.com/minio/minio/cmd/gateway/hdfs
cmd/gateway/hdfs/gateway-hdfs.go:223:9: cannot use &hdfsObjects literal (type *hdfsObjects) as type "github.com/minio/minio/cmd".ObjectLayer in return argument:
	*hdfsObjects does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
cmd/gateway/hdfs/gateway-hdfs.go:462:26: cannot use n (type *hdfsObjects) as type "github.com/minio/minio/cmd".ObjectLayer in argument to "github.com/minio/minio/cmd".ListObjects:
	*hdfsObjects does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
note: module requires Go 1.15
# github.com/minio/minio/cmd/gateway/azure
cmd/gateway/azure/gateway-azure.go:175:11: cannot use &azureObjects literal (type *azureObjects) as type "github.com/minio/minio/cmd".ObjectLayer in return argument:
	*azureObjects does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
cmd/gateway/azure/gateway-azure.go:177:10: cannot use &azureObjects literal (type *azureObjects) as type "github.com/minio/minio/cmd".ObjectLayer in return argument:
	*azureObjects does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
cmd/gateway/azure/gateway-azure.go:207:9: cannot use &azureObjects literal (type *azureObjects) as type "github.com/minio/minio/cmd".ObjectLayer in return argument:
	*azureObjects does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
note: module requires Go 1.15
# github.com/minio/minio/cmd/gateway/s3
cmd/gateway/s3/gateway-s3.go:248:10: cannot use &encS (type *s3EncObjects) as type "github.com/minio/minio/cmd".ObjectLayer in return argument:
	*s3EncObjects does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
cmd/gateway/s3/gateway-s3.go:250:9: cannot use &s (type *s3Objects) as type "github.com/minio/minio/cmd".ObjectLayer in return argument:
	*s3Objects does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
note: module requires Go 1.15
# github.com/minio/minio/cmd/gateway/gcs
cmd/gateway/gcs/gateway-gcs.go:208:2: cannot use gcs (type *gcsGateway) as type "github.com/minio/minio/cmd".ObjectLayer in return argument:
	*gcsGateway does not implement "github.com/minio/minio/cmd".ObjectLayer (wrong type for LocalStorageInfo method)
		have LocalStorageInfo() ("github.com/minio/minio/cmd".StorageInfo, []error)
		want LocalStorageInfo(context.Context) ("github.com/minio/minio/cmd".StorageInfo, []error)
note: module requires Go 1.15
make: *** [build] Error 2

```

--- a/cmd/admin-handlers.go
+++ b/cmd/admin-handlers.go
@@ -1551,13 +1551,13 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 	if globalLDAPConfig.Enabled {
 		ldapConn, err := globalLDAPConfig.Connect()
 		if err != nil {
-			ldap.Status = "offline"
+			ldap.Status = string(madmin.ItemOffline)
 		} else if ldapConn == nil {
 			ldap.Status = "Not Configured"
 		} else {
 			// Close ldap connection to avoid leaks.
 			ldapConn.Close()
-			ldap.Status = "online"
+			ldap.Status = string(madmin.ItemOnline)
 		}
 	}
 
@@ -1573,7 +1573,7 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 	assignPoolNumbers(servers)
 
 	var backend interface{}
-	mode := madmin.ObjectLayerInitializing
+	mode := madmin.ItemInitializing
 
 	buckets := madmin.Buckets{}
 	objects := madmin.Objects{}
@@ -1581,7 +1581,7 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 
 	objectAPI := newObjectLayerFn()
 	if objectAPI != nil {
-		mode = madmin.ObjectLayerOnline
+		mode = madmin.ItemOnline
 		// Load data usage
 		dataUsageInfo, err := loadDataUsageFromBackend(ctx, objectAPI)
 		if err == nil {
@@ -1628,7 +1628,7 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 	}
 
 	infoMsg := madmin.InfoMessage{
-		Mode:         mode,
+		Mode:         string(mode),
 		Domain:       domain,
 		Region:       globalServerRegion,
 		SQSARN:       globalNotificationSys.GetARNList(false),
@@ -1678,9 +1678,9 @@ func fetchLambdaInfo() []map[string][]madmin.TargetIDStatus {
 		active, _ := tgt.IsActive()
 		targetID := tgt.ID()
 		if active {
-			targetIDStatus[targetID.ID] = madmin.Status{Status: "Online"}
+			targetIDStatus[targetID.ID] = madmin.Status{Status: string(madmin.ItemOnline)}
 		} else {
-			targetIDStatus[targetID.ID] = madmin.Status{Status: "Offline"}
+			targetIDStatus[targetID.ID] = madmin.Status{Status: string(madmin.ItemOffline)}
 		}
 		list := lambdaMap[targetID.Name]
 		list = append(list, targetIDStatus)
@@ -1692,9 +1692,9 @@ func fetchLambdaInfo() []map[string][]madmin.TargetIDStatus {
 		active, _ := tgt.IsActive()
 		targetID := tgt.ID()
 		if active {
-			targetIDStatus[targetID.ID] = madmin.Status{Status: "Online"}
+			targetIDStatus[targetID.ID] = madmin.Status{Status: string(madmin.ItemOnline)}
 		} else {
-			targetIDStatus[targetID.ID] = madmin.Status{Status: "Offline"}
+			targetIDStatus[targetID.ID] = madmin.Status{Status: string(madmin.ItemOffline)}
 		}
 		list := lambdaMap[targetID.Name]
 		list = append(list, targetIDStatus)
@@ -1727,9 +1727,9 @@ func fetchKMSStatus() madmin.KMS {
 	}
 
 	if err := checkConnection(kmsInfo.Endpoints[0], 15*time.Second); err != nil {
-		kmsStat.Status = "offline"
+		kmsStat.Status = string(madmin.ItemOffline)
 	} else {
-		kmsStat.Status = "online"
+		kmsStat.Status = string(madmin.ItemOnline)
 
 		kmsContext := crypto.Context{"MinIO admin API": "ServerInfoHandler"} // Context for a test key operation
 		// 1. Generate a new key using the KMS.
@@ -1737,7 +1737,7 @@ func fetchKMSStatus() madmin.KMS {
 		if err != nil {
 			kmsStat.Encrypt = fmt.Sprintf("Encryption failed: %v", err)
 		} else {
-			kmsStat.Encrypt = "Ok"
+			kmsStat.Encrypt = "success"
 		}
 
 		// 2. Verify that we can indeed decrypt the (encrypted) key
@@ -1748,7 +1748,7 @@ func fetchKMSStatus() madmin.KMS {
 		case subtle.ConstantTimeCompare(key[:], decryptedKey[:]) != 1:
 			kmsStat.Decrypt = "Decryption failed: decrypted key does not match generated key"
 		default:
-			kmsStat.Decrypt = "Ok"
+			kmsStat.Decrypt = "success"
 		}
 	}
 	return kmsStat
@@ -1764,11 +1764,11 @@ func fetchLoggerInfo() ([]madmin.Logger, []madmin.Audit) {
 			err := checkConnection(target.Endpoint(), 15*time.Second)
 			if err == nil {
 				mapLog := make(map[string]madmin.Status)
-				mapLog[tgt] = madmin.Status{Status: "Online"}
+				mapLog[tgt] = madmin.Status{Status: string(madmin.ItemOnline)}
 				loggerInfo = append(loggerInfo, mapLog)
 			} else {
 				mapLog := make(map[string]madmin.Status)
-				mapLog[tgt] = madmin.Status{Status: "offline"}
+				mapLog[tgt] = madmin.Status{Status: string(madmin.ItemOffline)}
 				loggerInfo = append(loggerInfo, mapLog)
 			}
 		}
@@ -1780,11 +1780,11 @@ func fetchLoggerInfo() ([]madmin.Logger, []madmin.Audit) {
 			err := checkConnection(target.Endpoint(), 15*time.Second)
 			if err == nil {
 				mapAudit := make(map[string]madmin.Status)
-				mapAudit[tgt] = madmin.Status{Status: "Online"}
+				mapAudit[tgt] = madmin.Status{Status: string(madmin.ItemOnline)}
 				auditloggerInfo = append(auditloggerInfo, mapAudit)
 			} else {
 				mapAudit := make(map[string]madmin.Status)
-				mapAudit[tgt] = madmin.Status{Status: "Offline"}
+				mapAudit[tgt] = madmin.Status{Status: string(madmin.ItemOffline)}
 				auditloggerInfo = append(auditloggerInfo, mapAudit)
 			}
 		}

--- a/cmd/admin-server-info.go
+++ b/cmd/admin-server-info.go
@@ -42,16 +42,16 @@ func getLocalServerProperty(endpointServerPools EndpointServerPools, r *http.Req
 			}
 			if endpoint.IsLocal {
 				// Only proceed for local endpoints
-				network[nodeName] = "online"
+				network[nodeName] = string(madmin.ItemOnline)
 				localEndpoints = append(localEndpoints, endpoint)
 				continue
 			}
 			_, present := network[nodeName]
 			if !present {
 				if err := isServerResolvable(endpoint, 2*time.Second); err == nil {
-					network[nodeName] = "online"
+					network[nodeName] = string(madmin.ItemOnline)
 				} else {
-					network[nodeName] = "offline"
+					network[nodeName] = string(madmin.ItemOffline)
 					// log once the error
 					logger.LogOnceIf(context.Background(), err, nodeName)
 				}
@@ -59,35 +59,21 @@ func getLocalServerProperty(endpointServerPools EndpointServerPools, r *http.Req
 		}
 	}
 
-	localDisks, _ := initStorageDisksWithErrors(localEndpoints)
-	defer closeStorageDisks(localDisks)
-
-	storageInfo, _ := getStorageInfo(localDisks, localEndpoints.GetAllStrings())
-
-	return madmin.ServerProperties{
-		State:    "ok",
+	props := madmin.ServerProperties{
+		State:    string(madmin.ItemInitializing),
 		Endpoint: addr,
 		Uptime:   UTCNow().Unix() - globalBootTime.Unix(),
 		Version:  Version,
 		CommitID: CommitID,
 		Network:  network,
-		Disks:    storageInfo.Disks,
-	}
-}
-
-func getLocalDisks(endpointServerPools EndpointServerPools) []madmin.Disk {
-	var localEndpoints Endpoints
-	for _, ep := range endpointServerPools {
-		for _, endpoint := range ep.Endpoints {
-			if !endpoint.IsLocal {
-				continue
-			}
-			localEndpoints = append(localEndpoints, endpoint)
-		}
 	}
 
-	localDisks, _ := initStorageDisksWithErrors(localEndpoints)
-	defer closeStorageDisks(localDisks)
-	storageInfo, _ := getStorageInfo(localDisks, localEndpoints.GetAllStrings())
-	return storageInfo.Disks
+	objLayer := newObjectLayerFn()
+	if objLayer != nil {
+		storageInfo, _ := objLayer.LocalStorageInfo(GlobalContext)
+		props.State = string(madmin.ItemOnline)
+		props.Disks = storageInfo.Disks
+	}
+
+	return props
 }

--- a/cmd/erasure-common.go
+++ b/cmd/erasure-common.go
@@ -23,6 +23,16 @@ import (
 	"github.com/minio/minio/pkg/sync/errgroup"
 )
 
+func (er erasureObjects) getLocalDisks() (localDisks []StorageAPI) {
+	disks := er.getDisks()
+	for _, disk := range disks {
+		if disk != nil && disk.IsLocal() {
+			localDisks = append(localDisks, disk)
+		}
+	}
+	return localDisks
+}
+
 func (er erasureObjects) getLoadBalancedLocalDisks() (newDisks []StorageAPI) {
 	disks := er.getDisks()
 	// Based on the random shuffling return back randomized disks.

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -331,6 +331,35 @@ func (z *erasureServerPools) BackendInfo() (b BackendInfo) {
 	return
 }
 
+func (z *erasureServerPools) LocalStorageInfo(ctx context.Context) (StorageInfo, []error) {
+	var storageInfo StorageInfo
+
+	storageInfos := make([]StorageInfo, len(z.serverPools))
+	storageInfosErrs := make([][]error, len(z.serverPools))
+	g := errgroup.WithNErrs(len(z.serverPools))
+	for index := range z.serverPools {
+		index := index
+		g.Go(func() error {
+			storageInfos[index], storageInfosErrs[index] = z.serverPools[index].LocalStorageInfo(ctx)
+			return nil
+		}, index)
+	}
+
+	// Wait for the go routines.
+	g.Wait()
+
+	storageInfo.Backend = z.BackendInfo()
+	for _, lstorageInfo := range storageInfos {
+		storageInfo.Disks = append(storageInfo.Disks, lstorageInfo.Disks...)
+	}
+
+	var errs []error
+	for i := range z.serverPools {
+		errs = append(errs, storageInfosErrs[i]...)
+	}
+	return storageInfo, errs
+}
+
 func (z *erasureServerPools) StorageInfo(ctx context.Context) (StorageInfo, []error) {
 	var storageInfo StorageInfo
 

--- a/cmd/erasure-sets.go
+++ b/cmd/erasure-sets.go
@@ -601,6 +601,37 @@ func (s *erasureSets) StorageInfo(ctx context.Context) (StorageInfo, []error) {
 	return storageInfo, errs
 }
 
+// StorageInfo - combines output of StorageInfo across all erasure coded object sets.
+func (s *erasureSets) LocalStorageInfo(ctx context.Context) (StorageInfo, []error) {
+	var storageInfo StorageInfo
+
+	storageInfos := make([]StorageInfo, len(s.sets))
+	storageInfoErrs := make([][]error, len(s.sets))
+
+	g := errgroup.WithNErrs(len(s.sets))
+	for index := range s.sets {
+		index := index
+		g.Go(func() error {
+			storageInfos[index], storageInfoErrs[index] = s.sets[index].LocalStorageInfo(ctx)
+			return nil
+		}, index)
+	}
+
+	// Wait for the go routines.
+	g.Wait()
+
+	for _, lstorageInfo := range storageInfos {
+		storageInfo.Disks = append(storageInfo.Disks, lstorageInfo.Disks...)
+	}
+
+	var errs []error
+	for i := range s.sets {
+		errs = append(errs, storageInfoErrs[i]...)
+	}
+
+	return storageInfo, errs
+}
+
 // Shutdown shutsdown all erasure coded sets in parallel
 // returns error upon first error.
 func (s *erasureSets) Shutdown(ctx context.Context) error {

--- a/cmd/erasure.go
+++ b/cmd/erasure.go
@@ -229,6 +229,18 @@ func (er erasureObjects) StorageInfo(ctx context.Context) (StorageInfo, []error)
 	return getStorageInfo(disks, endpoints)
 }
 
+// LocalStorageInfo - returns underlying local storage statistics.
+func (er erasureObjects) LocalStorageInfo(ctx context.Context) (StorageInfo, []error) {
+	disks := er.getLocalDisks()
+	endpoints := make([]string, len(disks))
+	for i, disk := range disks {
+		if disk != nil {
+			endpoints[i] = disk.String()
+		}
+	}
+	return getStorageInfo(disks, endpoints)
+}
+
 func (er erasureObjects) getOnlineDisksWithHealing() (newDisks []StorageAPI, healing bool) {
 	var wg sync.WaitGroup
 	disks := er.getDisks()

--- a/cmd/fs-v1.go
+++ b/cmd/fs-v1.go
@@ -206,6 +206,11 @@ func (fs *FSObjects) BackendInfo() BackendInfo {
 	return BackendInfo{Type: BackendFS}
 }
 
+// LocalStorageInfo - returns underlying storage statistics.
+func (fs *FSObjects) LocalStorageInfo(ctx context.Context) (StorageInfo, []error) {
+	return fs.StorageInfo(ctx)
+}
+
 // StorageInfo - returns underlying storage statistics.
 func (fs *FSObjects) StorageInfo(ctx context.Context) (StorageInfo, []error) {
 	atomic.AddInt64(&fs.activeIOCount, 1)

--- a/cmd/gateway-unsupported.go
+++ b/cmd/gateway-unsupported.go
@@ -39,6 +39,13 @@ func (a GatewayUnsupported) BackendInfo() BackendInfo {
 	return BackendInfo{Type: BackendGateway}
 }
 
+// LocalStorageInfo returns the local disks information, mainly used
+// in prometheus - for gateway this just a no-op
+func (a GatewayUnsupported) LocalStorageInfo(ctx context.Context) (StorageInfo, []error) {
+	logger.CriticalIf(ctx, errors.New("not implemented"))
+	return StorageInfo{}, nil
+}
+
 // NSScanner - scanner is not implemented for gateway
 func (a GatewayUnsupported) NSScanner(ctx context.Context, bf *bloomFilter, updates chan<- DataUsageInfo) error {
 	logger.CriticalIf(ctx, errors.New("not implemented"))

--- a/cmd/gateway/hdfs/gateway-hdfs.go
+++ b/cmd/gateway/hdfs/gateway-hdfs.go
@@ -232,6 +232,10 @@ func (n *hdfsObjects) Shutdown(ctx context.Context) error {
 	return n.clnt.Close()
 }
 
+func (n *hdfsObjects) LocalStorageInfo(ctx context.Context) (si minio.StorageInfo, errs []error) {
+	return n.StorageInfo(ctx)
+}
+
 func (n *hdfsObjects) StorageInfo(ctx context.Context) (si minio.StorageInfo, errs []error) {
 	fsInfo, err := n.clnt.StatFs()
 	if err != nil {

--- a/cmd/metrics-v2.go
+++ b/cmd/metrics-v2.go
@@ -1023,7 +1023,7 @@ func getBucketUsageMetrics() MetricsGroup {
 		initialize: func(ctx context.Context, metrics *MetricsGroup) {
 			objLayer := newObjectLayerFn()
 			// Service not initialized yet
-			if objLayer == nil || globalIsGateway {
+			if objLayer == nil {
 				return
 			}
 
@@ -1031,7 +1031,7 @@ func getBucketUsageMetrics() MetricsGroup {
 				return
 			}
 
-			dataUsageInfo, err := loadDataUsageFromBackend(GlobalContext, objLayer)
+			dataUsageInfo, err := loadDataUsageFromBackend(ctx, objLayer)
 			if err != nil {
 				return
 			}
@@ -1040,8 +1040,8 @@ func getBucketUsageMetrics() MetricsGroup {
 			if dataUsageInfo.LastUpdate.IsZero() {
 				return
 			}
-			for bucket, usage := range dataUsageInfo.BucketsUsage {
 
+			for bucket, usage := range dataUsageInfo.BucketsUsage {
 				metrics.Metrics = append(metrics.Metrics, Metric{
 					Description:    getBucketUsageTotalBytesMD(),
 					Value:          float64(usage.Size),
@@ -1092,9 +1092,18 @@ func getLocalStorageMetrics() MetricsGroup {
 	return MetricsGroup{
 		Metrics: []Metric{},
 		initialize: func(ctx context.Context, metrics *MetricsGroup) {
-			disks := getLocalDisks(globalEndpoints)
-			for _, disk := range disks {
+			objLayer := newObjectLayerFn()
+			// Service not initialized yet
+			if objLayer == nil {
+				return
+			}
 
+			if globalIsGateway {
+				return
+			}
+
+			storageInfo, _ := objLayer.LocalStorageInfo(ctx)
+			for _, disk := range storageInfo.Disks {
 				metrics.Metrics = append(metrics.Metrics, Metric{
 					Description:    getNodeDiskUsedBytesMD(),
 					Value:          float64(disk.UsedSpace),
@@ -1120,15 +1129,18 @@ func getClusterStorageMetrics() MetricsGroup {
 	return MetricsGroup{
 		Metrics: []Metric{},
 		initialize: func(ctx context.Context, metrics *MetricsGroup) {
-
 			objLayer := newObjectLayerFn()
 			// Service not initialized yet
 			if objLayer == nil {
 				return
 			}
 
+			if globalIsGateway {
+				return
+			}
+
 			// Fetch disk space info, ignore errors
-			storageInfo, _ := objLayer.StorageInfo(GlobalContext)
+			storageInfo, _ := objLayer.StorageInfo(ctx)
 			onlineDisks, offlineDisks := getOnlineOfflineDisksStats(storageInfo.Disks)
 			totalDisks := offlineDisks.Merge(onlineDisks)
 
@@ -1142,15 +1154,14 @@ func getClusterStorageMetrics() MetricsGroup {
 				Value:       float64(GetTotalCapacityFree(storageInfo.Disks)),
 			})
 
-			s, _ := objLayer.StorageInfo(GlobalContext)
 			metrics.Metrics = append(metrics.Metrics, Metric{
 				Description: getClusterCapacityUsageBytesMD(),
-				Value:       GetTotalUsableCapacity(storageInfo.Disks, s),
+				Value:       GetTotalUsableCapacity(storageInfo.Disks, storageInfo),
 			})
 
 			metrics.Metrics = append(metrics.Metrics, Metric{
 				Description: getClusterCapacityUsageFreeBytesMD(),
-				Value:       GetTotalUsableCapacityFree(storageInfo.Disks, s),
+				Value:       GetTotalUsableCapacityFree(storageInfo.Disks, storageInfo),
 			})
 
 			metrics.Metrics = append(metrics.Metrics, Metric{

--- a/cmd/metrics.go
+++ b/cmd/metrics.go
@@ -521,6 +521,10 @@ func storageMetricsPrometheus(ch chan<- prometheus.Metric) {
 		return
 	}
 
+	if globalIsGateway {
+		return
+	}
+
 	server := getLocalServerProperty(globalEndpoints, &http.Request{
 		Host: GetLocalPeer(globalEndpoints),
 	})

--- a/cmd/notification.go
+++ b/cmd/notification.go
@@ -1231,9 +1231,9 @@ func (sys *NotificationSys) ServerInfo() []madmin.ServerProperties {
 			info, err := client.ServerInfo()
 			if err != nil {
 				info.Endpoint = client.host.String()
-				info.State = "offline"
+				info.State = string(madmin.ItemOffline)
 			} else {
-				info.State = "ok"
+				info.State = string(madmin.ItemOnline)
 			}
 			reply[idx] = info
 		}(client, i)
@@ -1306,7 +1306,7 @@ func GetPeerOnlineCount() (nodesOnline, nodesOffline int) {
 	nodesOffline = 0
 	servers := globalNotificationSys.ServerInfo()
 	for _, s := range servers {
-		if s.State == "ok" {
+		if s.State == string(madmin.ItemOnline) {
 			nodesOnline++
 			continue
 		}

--- a/cmd/object-api-interface.go
+++ b/cmd/object-api-interface.go
@@ -90,7 +90,8 @@ type ObjectLayer interface {
 	NSScanner(ctx context.Context, bf *bloomFilter, updates chan<- DataUsageInfo) error
 
 	BackendInfo() BackendInfo
-	StorageInfo(ctx context.Context) (StorageInfo, []error) // local queries only local disks
+	StorageInfo(ctx context.Context) (StorageInfo, []error)
+	LocalStorageInfo(ctx context.Context) (StorageInfo, []error)
 
 	// Bucket operations.
 	MakeBucketWithLocation(ctx context.Context, bucket string, opts BucketOptions) error

--- a/cmd/xl-storage.go
+++ b/cmd/xl-storage.go
@@ -295,7 +295,7 @@ func newXLStorage(ep Endpoint) (*xlStorage, error) {
 			return p, err
 		}
 		// error is unsupported disk, turn-off directIO for reads
-		logger.Info(fmt.Sprintf("Drive %s does not support O_DIRECT for reads, proceeding to use the drive without O_DIRECT", ep))
+		logger.LogOnceIf(GlobalContext, fmt.Errorf("Drive %s does not support O_DIRECT for reads, proceeding to use the drive without O_DIRECT", ep), ep.String())
 		p.readODirectSupported = false
 	}
 

--- a/pkg/madmin/info-commands.go
+++ b/pkg/madmin/info-commands.go
@@ -39,14 +39,17 @@ const (
 	// Add your own backend.
 )
 
-// ObjectLayerState - represents the status of the object layer
-type ObjectLayerState string
+// ItemState - represents the status of any item in offline,init,online state
+type ItemState string
 
 const (
-	// ObjectLayerInitializing indicates that the object layer is still in initialization phase
-	ObjectLayerInitializing = ObjectLayerState("initializing")
-	// ObjectLayerOnline indicates that the object layer is ready
-	ObjectLayerOnline = ObjectLayerState("online")
+
+	// ItemOffline indicates that the item is offline
+	ItemOffline = ItemState("offline")
+	// ItemInitializing indicates that the item is still in initialization phase
+	ItemInitializing = ItemState("initializing")
+	// ItemOnline indicates that the item is online
+	ItemOnline = ItemState("online")
 )
 
 // StorageInfo - represents total capacity of underlying storage.
@@ -171,7 +174,7 @@ func (adm *AdminClient) DataUsageInfo(ctx context.Context) (DataUsageInfo, error
 
 // InfoMessage container to hold server admin related information.
 type InfoMessage struct {
-	Mode         ObjectLayerState   `json:"mode,omitempty"`
+	Mode         string             `json:"mode,omitempty"`
 	Domain       []string           `json:"domain,omitempty"`
 	Region       string             `json:"region,omitempty"`
 	SQSARN       []string           `json:"sqsARN,omitempty"`

